sites/helpers.py

```diff
@@ -84,8 +82,19 @@ def safe_get(
     if headers:
         default_headers.update(headers)
     page = req.get(url, timeout=TIMEOUT_SECONDS, params=params, headers=default_headers)
+
+    # Many times, the request hits a 4xx or 5xx, but no exception is raised
+    # This makes sure an exception is raised and allows the retry decorator to work.
+    #
+    # Some notes:
+    # - Since we already have exponential backoff, no need to treat 429 error any differently.
+    # - During the last retry attempt, if this still raises an exception, that exception's already handled
+    # by a try-except block in each site's fetch_article method.
+    page.raise_for_status()
```

Comment:
❓ Do we know how this is handled if an exception is raised? I see `fetch_article` in each partner site has it wrapped in a `try/except`, but there are uses in `bulk_fetch` that aren't wrapped AFAICT.

Reply:
Thanks for pointing that out; didn't realize that and just assumed `bulk_fetch` would just use `fetch_article`.

Reply:
Wrapping these now.